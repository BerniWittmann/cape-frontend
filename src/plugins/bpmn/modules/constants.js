export const START_EVENT_TYPE = 'bpmn:StartEvent'
export const END_EVENT_TYPE = 'bpmn:EndEvent'
export const INTERMEDIATE_EVENT_TYPE = 'bpmn:IntermediateThrowEvent'
export const GATEWAY_TYPES = ['bpmn:ExclusiveGateway', 'bpmn:ParallelGateway', 'bpmn:InclusiveGateway', 'bpmn:ComplexGateway', 'bpmn:EventBasedGateway']
export const TASK_TYPES = ['bpmn:Task', 'bpmn:SendTask', 'bpmn:ReceiveTask', 'bpmn:UserTask', 'bpmn:ManualTask', 'bpmn:BusinessRuleTask', 'bpmn:ServiceTask', 'bpmn:ScriptTask', 'bpmn:CallActivity', 'bpmn:SubProcess']
export const DATA_OBJECT_TYPES = ['bpmn:DataObjectReference', 'bpmn:DataStoreReference']
export const MULTIPLE_SEQUENCE_FLOW_ALLOWED_OBJECT_TYPES = [...GATEWAY_TYPES, ...DATA_OBJECT_TYPES]
export const NO_CONNECTIONS_DISALLOWED_OBJECTS = [START_EVENT_TYPE, END_EVENT_TYPE, INTERMEDIATE_EVENT_TYPE, ...TASK_TYPES, ...GATEWAY_TYPES]
export const PROCESS_TYPES = ['bpmn:Process', 'bpmn:SubProcess']
