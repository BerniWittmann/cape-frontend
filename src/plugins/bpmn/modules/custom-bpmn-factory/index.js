import CustomBPMNFactory from './CustomBPMNFactory'

export default {
  __init__: [
    'bpmnFactory'
  ],
  bpmnFactory: [ 'type', CustomBPMNFactory ]
}
