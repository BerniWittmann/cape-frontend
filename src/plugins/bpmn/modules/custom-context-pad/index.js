import CustomContextPadProvider from './customContextPadProvider'

export default {
  __init__: [ 'contextPadProvider' ],
  contextPadProvider: [ 'type', CustomContextPadProvider ]
}
