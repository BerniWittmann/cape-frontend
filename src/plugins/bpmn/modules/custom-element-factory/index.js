import CustomElementFactory from './customElementFactory'

export default {
  __init__: [
    'elementFactory'
  ],
  elementFactory: [ 'type', CustomElementFactory ]
}
