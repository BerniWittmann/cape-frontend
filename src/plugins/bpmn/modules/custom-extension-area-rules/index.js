import CustomRules from './customRules'

export default {
  __init__: [
    'customExtensionAreaRules'
  ],
  customExtensionAreaRules: [ 'type', CustomRules ]
}
