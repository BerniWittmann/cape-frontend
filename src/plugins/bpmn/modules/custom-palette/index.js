import CustomPaletteProvider from './customPaletteProvider'

export default {
  __init__: [ 'paletteProvider' ],
  paletteProvider: [ 'type', CustomPaletteProvider ]
}
