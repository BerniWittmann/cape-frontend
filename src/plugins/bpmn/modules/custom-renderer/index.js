import CustomRenderer from './customRenderer'

export default {
  __init__: [
    'customRenderer'
  ],
  customRenderer: [ 'type', CustomRenderer ]
}
