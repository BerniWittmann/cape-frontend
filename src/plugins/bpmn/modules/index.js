import propertiesPanelModule from 'bpmn-js-properties-panel'
import propertiesPanelProvider from 'bpmn-js-properties-panel/lib/provider/camunda'

import customRules from './custom-rules'
import customContextPad from './custom-context-pad'
import customPalette from './custom-palette'
import customReplaceMenu from './custom-replace-menu'

export default [
  propertiesPanelModule,
  propertiesPanelProvider,
  customRules,
  customContextPad,
  customPalette,
  customReplaceMenu
]
