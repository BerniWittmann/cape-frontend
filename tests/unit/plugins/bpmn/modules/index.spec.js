import modules from '@/plugins/bpmn/modules'

jest.mock('bpmn-js-properties-panel', () => ({ name: 'properties-panel' }))
jest.mock('bpmn-js-properties-panel/lib/provider/camunda', () => ({ name: 'camunda-properties-panel' }))
jest.mock('@/plugins/bpmn/modules/custom-rules', () => ({ name: 'custom-rules' }))
jest.mock('@/plugins/bpmn/modules/custom-context-pad', () => ({ name: 'custom-context-pad' }))
jest.mock('@/plugins/bpmn/modules/custom-palette', () => ({ name: 'custom-palette' }))
jest.mock('@/plugins/bpmn/modules/custom-replace-menu', () => ({ name: 'custom-replace-menu' }))

describe('Plugins', () => {
  describe('BPMN', () => {
    describe('Modules', () => {
      it('exports an array', () => {
        expect(modules).toEqual(expect.any(Array))
      })

      it('has the correct configuration Keys', () => {
        expect(modules).toMatchSnapshot()
      })
    })
  })
})
