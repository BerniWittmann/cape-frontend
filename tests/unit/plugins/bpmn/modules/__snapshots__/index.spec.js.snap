// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugins BPMN Modules has the correct configuration Keys 1`] = `
Array [
  Object {
    "name": "properties-panel",
  },
  Object {
    "name": "camunda-properties-panel",
  },
  Object {
    "name": "custom-rules",
  },
  Object {
    "name": "custom-context-pad",
  },
  Object {
    "name": "custom-palette",
  },
  Object {
    "name": "custom-replace-menu",
  },
]
`;
